src/number.cpp

```diff
@@ -178,6 +178,50 @@ double eval_dbl(const number &n, const std::unordered_map<std::string, double> &
     return std::visit([](const auto &v) { return static_cast<double>(v); }, n.value());
 }
 
+void eval_batch_dbl(const number &n, const std::unordered_map<std::string, std::vector<double>> &,
+                    std::vector<double> &out_values)
+{
+    return std::visit(
+        [&out_values](const auto &v) {
+            for (auto &el : out_values) {
+                el = static_cast<double>(v);
+            }
+        },
+        n.value());
+}
+
+void update_connections(const number &, std::vector<std::vector<unsigned>> &node_connections, unsigned &node_counter)
+{
+    node_connections.push_back(std::vector<unsigned>());
+    node_counter++;
+}
+
+void update_node_values_dbl(const number &n, const std::unordered_map<std::string, double> &map,
+                            std::vector<double> &node_values,
+                            const std::vector<std::vector<unsigned>> &node_connections, unsigned &node_counter)
+{
+
+    std::visit([&node_values, &node_counter](const auto &v) { node_values[node_counter] = static_cast<double>(v); },
+               n.value());
+    node_counter++;
+}
+
+void update_grad_dbl(const number &, const std::unordered_map<std::string, double> &,
+                     std::unordered_map<std::string, double> &, const std::vector<double> &,
+                     const std::vector<std::vector<unsigned>> &, unsigned &node_counter, double)
+{
+    node_counter++;
+}
+
+// NOTE: for the generation of constants of other floating-point types
```

Comment:
Can you remove this block of comments? I think they are a leftover from a merge.